File: server/tools.js
exports.tools = {
  zoom_in: {
    id: 'zoom_in',
    description: 'This tool zooms in on the map',
    input: {},
  },
  zoom_out: {
    id: 'zoom_out',
    description: 'This tool zooms out on the map',
    input: {},
  },
  zoom_in_location: {
    id: 'zoom_in_location',
    description: 'This tool zooms in on the map to a specific location',
    input: {
      location: {
        type: 'string',
        description: 'The location (place) to zoom in on',
      },
    },
  },
  start_measuring: {
    id: 'start_measuring',
    description:
      'start the measuring scale on the map which measures distance between two points',
    input: {},
  },
  stop_measuring: {
    id: 'stop_measuring',
    description: 'stop the measuring scale on the map',
    input: {},
  },
  undo_measurement: {
    id: 'undo_measurement',
    description: 'undo the last measurement',
    input: {},
  },
  clear_measurement: {
    id: 'clear_measurement',
    description: 'clear all measurements',
    input: {},
  },
  start_marker_addition: {
    id: 'start_marker_addition',
    description: 'start adding markers to the map',
    input: {},
  },
  stop_marker_addition: {
    id: 'stop_marker_addition',
    description: 'stop adding markers to the map',
    input: {},
  },
  undo_marker: {
    id: 'undo_marker',
    description: 'undo the last marker addition',
    input: {},
  },
  clear_markers: {
    id: 'clear_markers',
    description: 'clear all markers',
    input: {},
  },
  search: {
    id: 'search',
    description: 'This tool searches for a specific location on the map',
    input: {
      location: {
        type: 'string',
        description: 'The location (place) to search for',
      },
    },
  },
  show_layer: {
    id: 'show_layers',
    description: 'show layers on the map',
    input: {
      layer: {
        type: 'string',
        description:
          'The layer to show enum - `OpenStreetMap|Search Results|Measures|Markers`',
      },
    },
  },
  hide_layer: {
    id: 'hide_layer',
    description: 'hide layers on the map',
    input: {
      layer: {
        type: 'string',
        description:
          'The layer to hide enum - `OpenStreetMap|Search Results|Measures|Markers`',
      },
    },
  },
};

File: nlp/src/tools.py
# tools.py

from enum import Enum

class Layer(Enum):
    OPENSTREETMAP = "OpenStreetMap"
    MEASURES = "Measures"
    MARKERS = "Markers"
    SEARCH = "Search Results"

tool_descriptions = {
    "zoom_in": "Zoom In - Increase map magnification, show more detail",
    "zoom_out": "Zoom Out - Decrease map magnification, show wider area",
    "zoom_in_location": "Zoom on Location - Focus and magnify specific location on map",
    "start_measuring": "Start measuring - Begin drawing lines to measure distances on map",
    "stop_measuring": "Stop / End distance measurement, return to normal map interaction",
    "undo_measurement": "Undo Measurement - Remove last segment of current distance measurement",
    "clear_measurement": "Clear Measurements - Remove all distance measurements from map",
    "search": "Search - Find and highlight specific location or point of interest",
    "start_marker_addition": "Start Marker Addition - Begin adding location markers to map",
    "stop_marker_addition": "Stop Marker Addition - End marker addition process",
    "undo_marker": "Undo Marker - Remove most recently added location marker",
    "clear_markers": "Clear Markers -  Remove all user-added location markers from map",
    "show_layer": "Show Layer - Display a specific map layer",
    "hide_layer": "Hide Layer - Hide a specific map layer"
}

layer_descriptions = {layer.value: f"show hide {layer.value} layer" for layer in Layer}
